File: 3737-count-subarrays-with-majority-element-i/3737-count-subarrays-with-majority-element-i.java
class Solution {
    public int countMajoritySubarrays(int[] nums, int target) {

        int len = nums.length;
        int[] prefixSum = new int[len];

        for(int i=0; i<len; i++) {
            int val = (nums[i] == target) ? 1 : -1;

            if (i==0) {
                prefixSum[i] = val;
            }else
                prefixSum[i] = prefixSum[i-1] + val;
        }   

        int majorityCount = 0;

        for (int i=0; i<len; i++) {
            for (int j=i; j<len; j++) {

                int sum;

                if (i == 0) {
                    sum = prefixSum[j];
                }else{
                    sum = prefixSum[j] - prefixSum[i-1];
                }

                if (sum > 0){
                    majorityCount++;
                }
            }
        }

        return majorityCount;
    }
}

File: 3737-count-subarrays-with-majority-element-i/3737-count-subarrays-with-majority-element-i.ts
function countMajoritySubarrays(nums: number[], target: number): number {
    
    let len: number = nums.length;
    let majorityCount: number = 0;

    for (let i=0; i<len; i++) {
        
        let targetCount: number = 0
        let nonTargetCount: number = 0;

        for(let j=i; j<len; j++) {

            if(nums[j] == target) {
                targetCount++;
            }else {
                nonTargetCount++;
            }

            if (targetCount > nonTargetCount) {
                majorityCount++;
            }
        }
    }

    return majorityCount;
};